Added basic syntax highlighting for k in nano

ColmBhandal opened the pull request:
Just highlighting some keywords & comments for the moment as nano's syntax highlighting seems to be pretty basic. Converted a bunch of keywords from the [Notepad++ syntax higlighter](https://github.com/kframework/k-editor-support/blob/master/notepad%2B%2B/k_syntax_highlight_nppp.xml).

The highlighting can certainly be extended I'm sure, but it's a start.

ColmBhandal commented:
> The highlighting can certainly be extended I'm sure, but it's a start.


After a few days hacking at this I'm not sure of the above & not sure if it makes sense to merge or not. I'll keep this open for now as I'm still using it myself, but welcome to hear any inputs. Problem is false positive highlighting. Fixed all examples I've encountered so far using regexes but I have a feeling I'm bound to hit a wall where regexes aren't enough. The questions then will be:

1. Does this get merged with a warning about false-positive highlighting in certain cases (some known, some unknown) & the limitations of regexes, and a note on how to turn it off or tweak it if the false-positives get annoying?
2. Or do we just leave it and let people using Nano edit in plain text without highlighting?

To me the pros of highlighting still outweigh the cons of the false positives, so I'm going to keep using this (at least most of the time) for now. But I can see both points of view.

I will note that this problem seems to be the issue with any nano attempt at syntax highlighting, so I guess at least it's par for the course. For example, the following code renders the text after ``//`` as a comment, even though it's inside a string, for the OOTB Nano Java syntax highlighting settings:

`String s = "Hel//lo";`

(GNU nano, version 3.2)

radumereuta commented:
I've updated the main keywords for K in this PR: https://github.com/kframework/k-editor-support/pull/20
If you update this PR to match the new format I can merge your changes as well.

ColmBhandal commented:
Thanks for the update @radumereuta. I've updated the keywords to match the latest updates to the Notepad++ ones.

What I will just flag, before this PR is merged, is that earlier I had added syntax highlighting around the XML-style tags to demarcate the cells. While I find this useful, I can see it as being annoying for some people. Here is an example - see the purple highlighting of the XML-style tags:

![image](https://user-images.githubusercontent.com/14222121/163942815-a5a1859f-6c6c-475b-bd49-519e26ea8a9d.png)

Of course, people can turn it off locally. But if you'd like me to get rid of this feature from the PR let me know.

Baltoli commented:
Is the background colour highlighting your **specific** editor configuration, or does that happen for every user? It's nice to have cells highlighted (see example here) but I think doing so with a background colour might not be to everyone's taste.

![Screenshot from 2022-04-19 09-41-38](https://user-images.githubusercontent.com/1329034/163962094-4b0b76dc-2b92-492b-a3d3-b3bd2282f8a4.png)

ColmBhandal commented:
@Baltoli I believe those background colours are going to appear for everyone else exactly as they appear for me. Let me explore using a foreground colour instead of a background colour and I'll post a screenshot here.

ColmBhandal commented:
@Baltoli I definitely prefer it now. I'm using brightyellow for the cell XML-tag highlighting:

![image](https://user-images.githubusercontent.com/14222121/163973590-ea739784-364f-4691-a838-68ef2c79ee72.png)

Baltoli commented:
Looks good to me.

radumereuta commented:
Can you also include `module` and `endmodule` in the list of keywords?
In the notepad++ config file they are special because they come in pairs.
Does nano have something similar? If not, just keywords should be fine though.

radumereuta commented on nano/k.nanorc:
```diff
@@ -0,0 +1,17 @@
+color cyan "\<(BUILTIN-SYNTAX-HOOKS|BUILTIN-HOOKS|SUBSTITUTION|PATTERN-MATCHING)\>"
+color cyan "\<(isMatching|getMatching|lengthString|chrChar|ordChar|substrString|subString|findSting|rfindString|String2Int|Int2String|categoryChar|directionalityChar|fresh|ListItem|SetItem|\.List|\.Set|\.K|\.|_)\>"
+color brightyellow start="<(\/)?(\w)+" end=">"
+color red "\<(module|endmodule|)\>"
```
I would suggest making them brightblue since they are keywords just like syntax, claim or rule...
